SYCL/ESIMD/api/functional/ctors/ctor_fill.hpp

```diff
@@ -0,0 +1,226 @@
+  }
+
+private:
+  template <typename DataT, int NumElems>
+  static void
+  call_simd_by_const_ref(const esimd::simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
+  }
+};
+
+// Enumeration of possible values for base value and step that will be provided
+// into simd constructor.
+enum class init_val {
+  min,
+  max,
+  zero,
+  min_half,
+  max_half,
+  neg_inf,
+  nan,
+  positive,
+  negative
+};
+
+// Dummy kernel for submitting some code into device side.
+template <typename DataT, int NumElems, typename T, init_val BaseVal,
+          init_val StepVal>
+struct kernel_for_fill;
+
+// Constructing a value for step and base values that depends on input
+// parameters.
+template <typename DataT, init_val BaseVal> DataT get_base_value() {
+  if constexpr (BaseVal == init_val::min) {
+    return value<DataT>::lowest();
+  } else if constexpr (BaseVal == init_val::max) {
+    return value<DataT>::max();
+  } else if constexpr (BaseVal == init_val::zero) {
+    return 0;
+  } else if constexpr (BaseVal == init_val::positive) {
+    return static_cast<DataT>(1.25);
+  } else if constexpr (BaseVal == init_val::negative) {
+    return static_cast<DataT>(-10.75);
+  } else if constexpr (BaseVal == init_val::min_half) {
+    return value<DataT>::lowest() / 2;
+  } else if constexpr (BaseVal == init_val::max_half) {
+    return value<DataT>::max() / 2;
+  } else if constexpr (BaseVal == init_val::neg_inf) {
+    return -value<DataT>::inf();
+  } else if constexpr (BaseVal == init_val::nan) {
+    return value<DataT>::nan();
+  } else {
+    static_assert(BaseVal != BaseVal, "Unexpected base value value");
+  }
+}
+
+// Struct that will be used for invocating of simd constructor.
+template <init_val BaseVal, init_val Step> struct call_simd_struct {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  esimd_tests::shared_vector<DataT>
+  call_simd(sycl::queue &queue, const std::vector<DataT> &ref_data) {
+    esimd_tests::shared_vector<DataT> result{NumElems,
+                                             shared_allocator<DataT>{queue}};
+    esimd_tests::shared_vector<DataT> shared_ref_data{
+        ref_data.begin(), ref_data.end(), shared_allocator<DataT>{queue}};
+    const auto ref = ref_data.data();
+    DataT step_value{get_base_value<DataT, Step>()};
+    DataT base_value{get_base_value<DataT, BaseVal>()};
+    queue.submit([&](sycl::handler &cgh) {
+      const auto ref = ref_data.data();
+      auto out = result.data();
+      cgh.single_task<
+          kernel_for_fill<DataT, NumElems, TestCaseT, BaseVal, Step>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(
+                base_value, step_value, out);
+          });
+    });
+
+    return result;
+  }
+};
+
+// Constructing verbose failure message.
+template <int NumElems>
+void on_failure(const std::string &step_value, const std::string &base_value,
```

Comment:
1 - Aren't you enforced to have `std::to_string()` repeated later in the code by using std::string here? Looks like you might to pass the values actually, and then just provide the generic way to "stringify" them
2 - Is the floating point string representation good here? Is it possible to provide some generic way to print fp-values with hex values printed (nan opcode for example)?

Reply:
I've applied this commit [here](https://github.com/intel/llvm-test-suite/pull/598/commits/652dcd3b7c72dceecf81bb7bd486398d8664c24d#diff-7fc90b589428e6e89076f07f5ff280a217b7f5c8e836bfe6c742734825074ef4R242-R244)

Reply:
That's definitely better, still we need to look into the test sources to have a hint about actual values used. Are we going to add hex representation later?

Reply:
> still we need to look into the test sources to have a hint about actual values used

Do you mean comment about what happens in the test? Yes, I am going to add it later

> Are we going to add hex representation later?

Yes, we have a plans for change representation of floating point types for printing it values to the console

Reply:
> Do you mean comment about what happens in the test?
Nope, I'm talking about console output on test failure. Still looks like you have a plenty of information printed already.

Reply:
console output will be looks like:
```
Failed for simd<unsigned short, 32>, with context: variable declaration, retrieved: 32767, expected: 65504, at index: 31, with base value: max_half, with step value: negative
```

@yuriykoch  Which information should be added?